### PR TITLE
[86b1qw3zj] Added Links & Routing to Navbar

### PR DESCRIPTION
**Pull request by ZoriTheDesigner**
# Type

`src`

# Summary

Added Links & Routing to Navbar, added basic fonts, icons, and styles.

# Changes

- Added logo, text, and shopping bag icon to navbar.
- Added links & routing.
- Embedded "The Four Seasons", and "Figtree" fonts from Adobe Fonts & Google Fonts
- Applied fonts to navbar.
- Adjusted spacing and responsiveness.(Desktop only)

### CODE REVIEW

**Comment by ZoriTheDesigner** [comment]
Task linked: [CU-86b1qw3zj Add Links & Routing to Navbar](https://app.clickup.com/t/86b1qw3zj)

File: src/ui/Navbar.fs [hunk 1]
```diff
@@ -1,6 +1,88 @@
...
+                            color "foreground"
+                            prop.onClick (fun _ -> Router.navigatePath "/products")
+                            prop.style [
+                                style.cursor "pointer"
+                            ]
+                        ]
+                    ]
+
+                    NavbarItem [
+                        prop.custom ("isActive", ())
```
Comment (wes-sleeman) [inline src/ui/Navbar.fs]: Looks like `isActive` isn't being correctly set here. You probably don't want to hardcode this. :)

File: src/ui/Navbar.fs [hunk 2]
```diff
@@ -1,6 +1,88 @@
...
+                        prop.children [
+                            Link [
+                                prop.text "About Us"
+                                prop.onClick (fun _ -> Router.navigatePath "/about")
+                                prop.style [
+                                style.cursor "pointer"
```
Comment (wes-sleeman) [inline src/ui/Navbar.fs]: This is the only link missing a `color` call. Is that on purpose?

File: src/ui/Navbar.fs [hunk 3]
```diff
@@ -1,6 +1,88 @@
...
+                            prop.style [
+                                style.cursor "pointer"
+                            ]
+                        ]
+                    ]
+                ]
+            ]
+
+            NavbarContent [
+                prop.custom ("justify", "end")
```
Comment (wes-sleeman) [inline src/ui/Navbar.fs]: I see a lot of these `prop.custom ("justify", ???)`. Maybe consider making a helper for it?

File: src/ui/Navbar.fs [hunk 4]
```diff
@@ -1,6 +1,88 @@
...
+                prop.children [
+                    NavbarItem [
+                        Button [
+                            color "primary"
+                            prop.href "#"
```
Comment (wes-sleeman) [inline src/ui/Navbar.fs]: `prop.href "#"` is likely a mistake as it doesn't actually do anything.

File: src/ui/NextUI.fs [hunk 5]
```diff
@@ -11,9 +11,43 @@ type NextUI =
         : ReactElement =
         React.imported ()
 
-    static member Component (name: string) (xs: IReactProperty list) : ReactElement =
-        Interop.reactApi.createElement (import name "@nextui-org/react", createObj !!xs)
+    static member inline color (colorName: string) = prop.custom("color", colorName)
```
Comment (wes-sleeman) [inline src/ui/NextUI.fs]: This is good. More like this as needed. I think any time we use `prop.custom` on the same key more than once, it should get one of these instead.

File: src/ui/NextUI.fs [hunk 6]
```diff
@@ -11,9 +11,43 @@ type NextUI =
...
     // Just change the function name from "Button" to whatever you need in both places.
-    static member inline Button: IReactProperty list -> ReactElement = NextUI.Component "Button"
+    static member inline Button(xs: IReactProperty list) : ReactElement =
+        Interop.reactApi.createElement (import "Button" "@nextui-org/react", createObj !!xs)
+
+    static member inline Navbar(xs: IReactProperty list) : ReactElement =
+        Interop.reactApi.createElement (import "Navbar" "@nextui-org/react", createObj !!xs)
+
+    static member inline Navbar(xs: ReactElement list) : ReactElement =
+        Interop.reactApi.createElement (import "Navbar" "@nextui-org/react", createObj !![ prop.children xs ])
```
Comment (wes-sleeman) [inline src/ui/NextUI.fs]: Could be interesting to make this a call to the other `Navbar` and reduce code duplication.
```suggestion
        Navbar [ prop.children xs ]
```

File: src/ui/NextUI.fs [hunk 7]
```diff
@@ -11,9 +11,43 @@ type NextUI =
...
+
+    static member inline NavbarBrand(xs: IReactProperty list) : ReactElement =
+        Interop.reactApi.createElement (import "NavbarBrand" "@nextui-org/react", createObj !!xs)
+
+    static member inline NavbarContent(xs: IReactProperty list) : ReactElement =
+        Interop.reactApi.createElement (import "NavbarContent" "@nextui-org/react", createObj !!xs)
+
+    static member inline NavbarContent(xs: ReactElement list) : ReactElement =
+        Interop.reactApi.createElement (import "NavbarContent" "@nextui-org/react", createObj !![ prop.children xs ])
```
Comment (wes-sleeman) [inline src/ui/NextUI.fs]:
```suggestion
        NavbarContent [ prop.children xs ]
```
As above.

File: src/ui/NextUI.fs [hunk 8]
```diff
@@ -11,9 +11,43 @@ type NextUI =
...
+
+    static member inline NavbarItem(xs: IReactProperty list) : ReactElement =
+        Interop.reactApi.createElement (import "NavbarItem" "@nextui-org/react", createObj !!xs)
+
+    static member inline NavbarItem(xs: ReactElement list) : ReactElement =
+        Interop.reactApi.createElement (import "NavbarItem" "@nextui-org/react", createObj !![ prop.children xs ])
```
Comment (wes-sleeman) [inline src/ui/NextUI.fs]:
```suggestion
        NavbarItem [ prop.children xs ]
```
Here too!

**Review by wes-sleeman — CHANGES_REQUESTED** [review]
Looks great! Awesome work here. Just a few minor changes and you're golden. :)

File: src/ui/Navbar.fs (hunk 1 above)
Comment (ZoriTheDesigner) [reply src/ui/Navbar.fs]: That was in NextUI's demo code. I removed it.

File: src/ui/Navbar.fs (hunk 2 above)
Comment (ZoriTheDesigner) [reply src/ui/Navbar.fs]: Nope. It has been added.

File: src/ui/Navbar.fs (hunk 3 above)
Comment (ZoriTheDesigner) [reply src/ui/Navbar.fs]: Added helper for justify, updated code using helper.

File: src/ui/Navbar.fs (hunk 4 above)
Comment (ZoriTheDesigner) [reply src/ui/Navbar.fs]: K.

File: src/ui/NextUI.fs (hunk 5 above)
Comment (ZoriTheDesigner) [reply src/ui/NextUI.fs]: Okay!

File: src/ui/NextUI.fs (hunk 6 above)
Comment (ZoriTheDesigner) [reply src/ui/NextUI.fs]: What do you mean?

Comment (ZoriTheDesigner) [reply src/ui/NextUI.fs]: Ah, okay.

Comment (wes-sleeman) [reply src/ui/NextUI.fs]: This one still needs doing. :)

**Review by wes-sleeman — CHANGES_REQUESTED** [review]
Almost there! Just one slight tweak.

**Review by wes-sleeman — APPROVED** [review]
Great work! Approved.

**Pull request merged**